rclcpp/include/rclcpp/wait_set_policies/static_storage.hpp

```diff
@@ -160,6 +160,15 @@ class StaticStorage : public rclcpp::wait_set_policies::detail::StoragePolicyCom
       services_,
       waitables_
     );
+
+    if(this->needs_pruning_) {
+      // we need to throw here, as the indexing of the rcl_waitset is broken,
+      // in case of invalid entries
+
+      throw std::runtime_error(
+            "StaticStorage : storage_rebuild_rcl_wait_set: Detected"
+            " invalid entity in static entity storage");
```

Comment:
Suggested edit, don't block the pr on this:

```suggestion
            "StaticStorage::storage_rebuild_rcl_wait_set(): entity weak_ptr "
            "unexpectedly expired in static entity storage");
```

Reply:
updated